sycl/include/sycl/ext/oneapi/experimental/sycl_complex.hpp

```diff
@@ -158,222 +157,32 @@ template <class _Tp> class complex {
-    __re_ = __re;
-    __im_ = value_type();
-    return *this;
-  }
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator+=(sycl::half __re) {
-    __re_ += __re;
-    return *this;
-  }
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator-=(sycl::half __re) {
-    __re_ -= __re;
-    return *this;
-  }
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator*=(sycl::half __re) {
-    __re_ *= __re;
-    __im_ *= __re;
-    return *this;
-  }
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator/=(sycl::half __re) {
-    __re_ /= __re;
-    __im_ /= __re;
-    return *this;
-  }
-
-  template <class _Xp>
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator=(const complex<_Xp> &__c) {
-    __re_ = __c.real();
-    __im_ = __c.imag();
-    return *this;
-  }
-  template <class _Xp>
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &
-  operator=(const std::complex<_Xp> &__c) {
-    __re_ = __c.real();
-    __im_ = __c.imag();
-    return *this;
-  }
-  template <class _Xp>
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &
-  operator+=(const complex<_Xp> &__c) {
-    __re_ += __c.real();
-    __im_ += __c.imag();
-    return *this;
-  }
-  template <class _Xp>
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &
-  operator-=(const complex<_Xp> &__c) {
-    __re_ -= __c.real();
-    __im_ -= __c.imag();
-    return *this;
-  }
-  template <class _Xp>
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &
-  operator*=(const complex<_Xp> &__c) {
-    *this = *this * complex(__c.real(), __c.imag());
-    return *this;
-  }
-  template <class _Xp>
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &
-  operator/=(const complex<_Xp> &__c) {
-    *this = *this / complex(__c.real(), __c.imag());
-    return *this;
-  }
-};
-
-template <> class complex<float> {
-  float __re_;
-  float __im_;
-
-public:
-  typedef float value_type;
-
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY constexpr complex(float __re = 0.0f,
-                                                     float __im = 0.0f)
-      : __re_(__re), __im_(__im) {}
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY
-  constexpr complex(const complex<sycl::half> &__c);
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY
-  explicit constexpr complex(const complex<double> &__c);
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY
-  constexpr complex(const std::complex<float> &__c)
-      : __re_(__c.real()), __im_(__c.imag()) {}
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY
-  constexpr operator std::complex<float>() {
-    return std::complex<float>(__re_, __im_);
-  }
-
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY constexpr float real() const {
-    return __re_;
-  }
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY constexpr float imag() const {
-    return __im_;
-  }
-
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY void real(value_type __re) { __re_ = __re; }
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY void imag(value_type __im) { __im_ = __im; }
-
-  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator=(float __re) {
+  _SYCL_EXT_CPLX_INLINE_VISIBILITY complex &operator=(value_type __re) {
```

Comment:
Could you please explain what the benefit is to change it to a copy rather than a constant reference?

Reply:
For an "in" parameter, passing cheaply-copied types by value rather than constant reference is good practice.
More overhead on access if it was `const value_type &__re`

Also, I might be misinterpreting your comment, but your phrasing indicates that these functions were receiving their argument by constant ref which was never the case.

Please correct me if I misunderstood your comment :smile:

Reply:
I suppose since `value_type` is always cheaply copied, that should be fine. Thanks!